var/spack/repos/builtin/packages/sensei/package.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Sensei(CMakePackage):
+    """SENSEI is a platform for scalable in-situ analysis and visualization.
+    Its design motto is 'Write once, run everywhere', this means that once
+    the application is instrumented with SENSEI it can use existing and
+    future analysis backends. Existing backends include: Paraview/Catalyst,
+    Visit/Libsim, ADIOS, Python scripts, and so on."""
+
+    homepage = "https://sensei-insitu.org"
+    url      = "https://gitlab.kitware.com/sensei/sensei/-/archive/v2.1.1/sensei-v2.1.1.tar.gz"
+    git      = "https://gitlab.kitware.com/sensei/sensei.git"
+
+    version('master', branch='master')
+    version('2.1.1', sha256='8a27ebf133fef00a59e4b29433762e6560bf20214072de7808836eb668bb5687')
+    version('2.1.0', sha256='b7af21a25523cf6cd8934d797471b75ca32881166625d71f24b5c8b6d727ca99')
+    version('2.0.0', sha256='df48eab035e1acdd8edf5159955c05306f9ca48117effacc4a6b77c3fb24f62b')
+    version('1.1.0', sha256='e5a4ba691573ff6c7b0d4793665e218ee5868ebcc0198915d1f16a4b7b92a368')
+    version('1.0.0', sha256='bdcb03c56b51f2795ec5a7e85a5abb01d473d192fac50f2a8bf2608cc3564ff8')
+
+    variant('sencore', default=True, description='Enables the SENSEI core library')
+    variant('catalyst', default=True, description='Build with ParaView-Catalyst support')
+    variant('libsim', default=False, description='Build with VisIt-Libsim support')
+    variant('vtkio', default=True, description='Enable adaptors to write to VTK XML format')
+    variant('adios', default=False, description='Enable ADIOS adaptors and endpoints')
+    variant('python', default=False, description='Enable Python bindings')
+    variant('miniapps', default=True, description='Enable the parallel 3D and oscillators miniapps')
+
+    # Paraview 5.6 depends on Python 3, but all SENSEI versions up to 2.2.1 do
+    # not support Python 3. So for now, until SENSEI 3, is available SENSEI
+    # can only work with Paraview 5.5.x
+    depends_on("paraview@5.5.0:5.5.2+python+mpi+hdf5", when="+catalyst")
+    depends_on("visit", when="+libsim")
+    depends_on("vtk", when="+libsim")
+    depends_on("vtk", when="~libsim ~catalyst")
+    depends_on("adios", when="+adios")
+    # SENSEI does not support Python 3 for now. Once SENSEI 3 is available, the
+    # line below should be modified
+    depends_on("python@:2", when="+python")
+    depends_on("py-numpy", when="+python")
+    depends_on("py-mpi4py", when="+python")
+    depends_on("swig", when="+python")
+
+    # Can have either LibSim or Catalyst, but not both
+    conflicts('+libsim', when='+catalyst')
+
+    def cmake_args(self):
+        spec = self.spec
+
+        # -Ox flags are set by default in CMake based on the build type
+        args = [
+            '-DCMAKE_CXX_STANDARD=11',
```

Comment:
In case that you are not requiring exactly C++11, we usually expose this as a package variant.
```py
    variant('cxxstd',
            default='11',
            values=('11', '14', '17'),
            multi=False,
            description='Use the specified C++ standard when building.')
```
and use e.g. via
```suggestion
            '-DCMAKE_CXX_STANDARD={0}'.format(spec.variants['cxxstd'].value),
```

Reply:
This is a good suggestion, I'll add it